lightning/src/ln/channel.rs

```diff
@@ -4480,6 +4759,105 @@ impl<SP: Deref> Channel<SP> where
 		Ok(())
 	}
 
+	pub fn commitment_signed_initial_v2<L: Deref>(
```

Comment:
Here we are just adding it for inbound V2 channels, but actually, eventually outbound will also need it once that's enabled (https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#the-commitment_signed-message), and I'll probably share the behaviour via `InteractivelyFunded` trait or something like that.

Reply:
Ok so after looking at this for a while, it's something we can share behaviour for via the template method pattern Jeff suggested and have on `OutboundV2Channel` impl to align with `funding_signed` in `OutboundV1Channel` and similarly for `InboundV2Channel/InboundV1Channel`. However, this is under the assumption that there is a similar ordering to the sending of the initial `commitment_signed`.

For V1, exchange of commitment sigs is always strictly ordered:
Initiator followed by acceptor as `funding_signed` strictly follows `funding_created`.

For V2, there is no prescribed order of sending `commitment_signed`s, so the initiator or acceptor could technically send theirs first.

As acceptor we would by no means violate the spec if we just assume the channel initiator will send `commitment_signed` first and then we follow after that and then promote `InboundV2Channel` to a `Channel`. This would be like acceptor getting `funding_created` in the V1 case.

If we are the initiator, from our point of view immediately following consecutive `tx_complete` msgs, we queue a `commitment_signed` and promote `OutboundV2Channel` to a `Channel`.

With those reasonable assumptions, we can keep things in V2 establishment analogous to V1 for initial commitment signature exchange?

Reply:
Fair enough around no new assumptions and this can definitely be DRYed up in a prefactor commit!